src/components/NumberInput.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {ChangeEvent, useState} from "react"
+import {InputProps} from "../interface/InputProps.ts"
+
+const NumberInput: React.FC<InputProps> = ({
+                                               type,
+                                               className,
+                                               maxLength,
+                                               placeHolder,
+                                               inputRule,
+                                               inputChange,
+                                               disabled = false,
+                                               defaultState= ''
+                                           }) => {
+    const [inputNumber, setInputNumber] = useState<string | ''>(defaultState)
```

Comment:
r: string에 ''가 포함 관계라 string만 명시하는 것과 동일합니다.
```suggestion
    const [inputNumber, setInputNumber] = useState<string>(defaultState)
```

Reply:
수정했습니다.